Rakefile

```diff
@@ -3,3 +3,30 @@
 require 'bundler/setup'
 
 require 'bundler/gem_tasks'
+
+desc 'Prepare a new version for release, version can be major, minor, patch or x.y.z (as per gem-release gem)'
+task :prepare_release, %i[version] do |_, args|
+  bump_version = args.fetch(:version)
+
+  current_branch = `git branch --show-current`.chomp
+  raise 'could not get current branch' if current_branch.empty?
+
+  sh 'git', 'checkout', '-b', 'new-release' if current_branch == 'main'
```

Comment:
Will this fail the second time unless we delete the branch locally? Maybe we should name the branch after the version?

Reply:
Ya, so that's tricky because we don't know the version until the next command ... we could use `gem bump --pretend` and parse the output, I considered that, but went with simple.

wrt failing the second time, the default when we merge a PR is to delete the branch, so I assumed this would actually not be an issue in practice. Of course, that risks completely backfiring ... wdyt? Worth the extra complexity of trying to extract the new version number?

Reply:
Oh, the other option, of course, is bump as we do, without committing, and then creating the branch after. I was worried that creating the branch might fail, for reasons unknown, and we'd be straight back to messy-land.